File: 0139-word-break/0139-word-break.py
class Solution:
    def wordBreak(self, s: str, wordDict: List[str]) -> bool:
        dp = [False] * (len(s)+1)
        dp[0] = True
        for i in range(0, len(s)+1):
            for word in wordDict:
                if dp[i] and s[i : i+len(word)] == word:
                    dp[i+len(word)]= True
        return dp[len(s)]

File: 0139-word-break/0139-word-break.java
class Solution {
    public boolean wordBreak(String s, List<String> wordDict) {
        boolean[] dp = new boolean[s.length()+1];
        dp[0] = true;
        for(int i=0; i<=s.length(); i++){
            for(String str : wordDict){
                if(dp[i] && i+str.length()<=s.length() && s.substring(i, i+str.length()).equals(str)){
                    dp[i+str.length()] = true;
                }
            }
        }
        return dp[s.length()];
    }
}